x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/setup_knowledge_base_button.tsx

```diff
@@ -48,19 +52,35 @@ export const SetupKnowledgeBaseButton: React.FC = React.memo(() => {
 
   return (
     <EuiToolTip position={'bottom'} content={toolTipContent}>
-      <EuiButton
-        color="primary"
-        data-test-subj="setup-knowledge-base-button"
-        fill
-        disabled={!kbStatus?.is_setup_available}
-        isLoading={isSetupInProgress}
-        iconType="importAction"
-        onClick={onInstallKnowledgeBase}
-      >
-        {i18n.translate('xpack.elasticAssistant.knowledgeBase.installKnowledgeBaseButton', {
-          defaultMessage: 'Setup Knowledge Base',
-        })}
-      </EuiButton>
+      {display === 'mini' ? (
+        <EuiButtonEmpty
+          color="primary"
+          data-test-subj="setup-knowledge-base-button"
+          disabled={!kbStatus?.is_setup_available}
+          isLoading={isSetupInProgress}
+          iconType="importAction"
+          onClick={onInstallKnowledgeBase}
+          size={'xs'}
+        >
+          {i18n.translate('xpack.elasticAssistant.knowledgeBase.installKnowledgeBaseButton', {
+            defaultMessage: 'Setup Knowledge Base',
+          })}
+        </EuiButtonEmpty>
+      ) : (
+        <EuiButton
+          color="primary"
+          data-test-subj="setup-knowledge-base-button"
+          fill
+          disabled={!kbStatus?.is_setup_available}
```

Comment:
Consider hiding the button after setup completes. The screenshot below illustrates the button still visible after setup:

![setup_kb](https://github.com/user-attachments/assets/14cd838b-97d0-4fba-8558-68acc1fd5071)

Reply:
Paired and confirmed this was an artifact from the old mappings. Resolved once updated to V2 mappings.